dotnet-authserver/src/TeacherIdentity.AuthServer/DqtApiClient.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Net.Http.Headers;
+using TeacherIdentity.AuthServer.Models;
+
+namespace TeacherIdentity.AuthServer;
+
+public class DqtApiClient : IDqtApiClient
+{
+    private IConfiguration _configuration;
+
+    public DqtApiClient(IConfiguration configuration)
+    {
+        _configuration = configuration;
+    }
+
+    private HttpClient GetHttpClient()
+    {
+        var client = new HttpClient();
+        var apiSecret = _configuration.GetSection("DqtApi").GetValue<string>("apiSecret");
+        var apiUrl = _configuration.GetSection("DqtApi").GetValue<string>("ApiUrl");
+        client.BaseAddress = new Uri(apiUrl);
+        client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", apiSecret);
+        return client;
+    }
+
+    public async Task<TeacherIdentityInfo?> GetTeacherIdentityInfoAsync(string userId)
+    {
+        var client = GetHttpClient();
+        var response = await client.GetFromJsonAsync<TeacherIdentityInfo?>($"/v2/teachers/teacher-identity?tsPersonId={userId}");
+        return response;
+    }
+
+    public async Task PutTeacherIdentityInfo(TeacherIdentityInfo? info)
+    {
+        var client = GetHttpClient();
+        var response = await client.PutAsJsonAsync($"/v2/teachers/teacher-identity/{info?.Trn}", info);
```

Comment:
Body should only contain the `TsPersonId` parameter
```suggestion
        var response = await client.PutAsJsonAsync($"/v2/teachers/teacher-identity/{info?.Trn}", new { TsPersonId = info.TsPersonId });
```